examples/flax/token-classification/run_ner.py

```diff
@@ -0,0 +1,583 @@
+        return state.logits_fn(logits)
+
+    p_eval_step = jax.pmap(eval_step, axis_name="batch")
+
+    metric = load_metric("seqeval")
+
+    def get_labels(y_pred, y_true):
+        # Transform predictions and references tensos to numpy arrays
+
+        # Remove ignored index (special tokens)
+        true_predictions = [
+            [label_list[p] for (p, l) in zip(pred, gold_label) if l != -100]
+            for pred, gold_label in zip(y_pred, y_true)
+        ]
+        true_labels = [
+            [label_list[l] for (p, l) in zip(pred, gold_label) if l != -100]
+            for pred, gold_label in zip(y_pred, y_true)
+        ]
+        return true_predictions, true_labels
+
+    def compute_metrics():
+        results = metric.compute()
+        if args.return_entity_level_metrics:
+            # Unpack nested dictionaries
+            final_results = {}
+            for key, value in results.items():
+                if isinstance(value, dict):
+                    for n, v in value.items():
+                        final_results[f"{key}_{n}"] = v
+                else:
+                    final_results[key] = value
+            return final_results
+        else:
+            return {
+                "precision": results["overall_precision"],
+                "recall": results["overall_recall"],
+                "f1": results["overall_f1"],
+                "accuracy": results["overall_accuracy"],
+            }
+
+    logger.info(f"===== Starting training ({num_epochs} epochs) =====")
+    train_time = 0
+
+    # make sure weights are replicated on each device
+    state = replicate(state)
+
+    for epoch in range(1, num_epochs + 1):
+        logger.info(f"Epoch {epoch}")
+        logger.info("  Training...")
+
+        train_start = time.time()
+        train_metrics = []
+        rng, input_rng = jax.random.split(rng)
+
+        # train
+        for batch in train_data_collator(input_rng, train_dataset, train_batch_size):
+            state, metrics, dropout_rngs = p_train_step(state, batch, dropout_rngs)
+            train_metrics.append(metrics)
+
+        train_time += time.time() - train_start
+        logger.info(f"    Done! Training metrics: {unreplicate(metrics)}")
+
+        logger.info("  Evaluating...")
+
+        # evaluate
+        for batch in eval_data_collator(eval_dataset, eval_batch_size):
+            labels = batch.pop("labels")
+            predictions = p_eval_step(state, batch)
+            predictions = jnp.array([pred for pred in chain(*predictions)])
+            labels = np.array([label for label in chain(*labels)])
+            labels[np.array(chain(*batch["attention_mask"])) == 0] = -100
+            preds, refs = get_labels(predictions, labels)
+            metric.add_batch(
+                predictions=preds,
+                references=refs,
+            )
+
+        # evaluate also on leftover examples (not divisible by batch_size)
```

Comment:
(nit), we should also enable step-wise logging, evaluation, and save (which helps avoid memory fragmentation in JAX) to be consistent with other flax examples. You could refer to `run_flax_mlm` or `run_flax_clm` to save how it's used.

Reply:
done